IGNITE-22278 Sql. Refactor base and planning contexts

korlov42 opened the pull request:
https://issues.apache.org/jira/browse/IGNITE-22278

This patch refactors contexts used within sql engine:
1) BaseQueryContext is removed. Instead lightweight SqlOperationContext is introduced. All planning-related metadata moved to PlanningContext
2) PrepareService is not aware of SchemaManager. It accepts operation time as input and resolve active schema itself
3) `schemaVersion` renamed to `catalogVersion` in numerous places to better reflect semantic

-------------------

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

lowka commented:
There are few places in sql-engine module where `schemaVersion` is still being used:

- `KeyValueModifyPlan` has a field `schemaVersion`
- `KeyValueGetPlan` has a field `schemaVersion`
- `SqlSchemaManager` defines a method `table(schemaVersion, tableId)`
- `ExecutionServiceImpl FragmentCacheKey` has a field `schemaVersion`

Are the names of these fields/variables kept unchanged on purpose?

korlov42 commented:
@lowka my bad, thanks for noticing. I didn't use full text search, just changed names in those places I stumbled upon.

Renamed also mentioned places.

xtern commented on modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/IgniteSchema.java:
```diff
@@ -44,7 +44,7 @@ public class IgniteSchema extends AbstractSchema {
     public IgniteSchema(String name, int version, Collection<? extends IgniteDataSource> tables) {
         this.name = name;
-        this.version = version;
+        this.catalogVersion = version;
```
Perhaps it makes sense to rename the parameter too?
(and the SCHEMA_VERSION constant in the ExecutionServiceImplTest test)

korlov42 replied:
done

xtern commented on modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManager.java:
```diff
@@ -25,23 +25,23 @@
-     * Returns root schema of the given version.
+     * Returns root schema derived from catalog of the given version.
      */
     SchemaPlus schema(int version);
```
```suggestion
    SchemaPlus schema(int catalogVersion);
```

korlov42 replied:
renamed

xtern commented on modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PrepareServiceImpl.java:
```diff
@@ -126,7 +133,8 @@ public static PrepareServiceImpl create(
             SqlDistributedConfiguration clusterCfg,
-            SqlLocalConfiguration nodeCfg
+            SqlLocalConfiguration nodeCfg,
+            SqlSchemaManager schemaManager
```
javadoc is missing :sunglasses:

korlov42 replied:
fixed, thanks!

xtern commented on modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/PrepareServiceImpl.java:
```diff
@@ -156,17 +165,18 @@ public PrepareServiceImpl(
             int plannerThreadCount,
-            MetricManager metricManager
+            MetricManager metricManager,
+            SqlSchemaManager schemaManager
```
javadoc is missing

korlov42 replied:
done
